plugins/carla/CMakeLists.txt

```diff
@@ -44,6 +44,15 @@ target_sources(
           pluginrefreshdialog.hpp
           qtutils.cpp)
 
+# non-mac/win systems should use rpath. this is typically automatically handled by pkg-config, but OBS does not read the
+# link flags from there
+if(NOT (OS_MACOS OR OS_WINDOWS))
+  set_property(
+    TARGET carla-bridge
+    APPEND
+    PROPERTY BUILD_RPATH "$<TARGET_FILE_DIR:carla::utils>")
```

Comment:
The `RPATH` will only exist in the build tree, but removed once the target is `install`ed (for that you'd have to set a `INSTALL_RPATH` iirc).

Using `RPATH` in installed targets will make the plugin invalid on specific Linux distributions which outright prohibit use of any `RPATH` hints in a binary though.

Reply:
right, I need to install the `INSTALL_RPATH` here too.

I dont see how this blocks things, carla requires rpath already so any distro packaging carla will already have that in place. the rpath is directly part of the pkg-config file as can be seen here https://github.com/falkTX/Carla/blob/main/data/pkgconfig/carla-utils.pc#L8
This special exception is needed because the OBS cmake way of importing libs does not make use of any extra linker options provided by that pkg-config file, so we have to force it here.

Reply:
Fedora maintainers will probably not be able to generate an `obs-studio` package with Carla with custom RPATHs used, if I understood this correctly:

https://fedoraproject.org/wiki/Changes/Broken_RPATH_will_fail_rpmbuild

Reply:
I wouldn't worry about it, they specifically mention *broken* RPATH which is not the case here.

Fedora already ships with LMMS with the Carla plugin enabled https://src.fedoraproject.org/rpms/lmms/blob/rawhide/f/lmms.spec#_76 so there is precedence for allowing this too.

Reply:
> Using `RPATH` in installed targets will make the plugin invalid on specific Linux distributions which outright prohibit use of any `RPATH` hints in a binary though.

They don't prohibit it, ti just Fedora doing they're own rules.